lib/custom_dropdown.dart

```diff
@@ -540,12 +541,18 @@ class _CustomDropdownState<T> extends State<CustomDropdown<T>> {
   void didUpdateWidget(covariant CustomDropdown<T> oldWidget) {
     super.didUpdateWidget(oldWidget);
 
-    if (widget.initialItem != oldWidget.initialItem) {
-      selectedItemNotifier = SingleSelectController(widget.initialItem);
+    if (widget.initialItem != oldWidget.initialItem &&
+        selectedItemNotifier.value != widget.initialItem) {
+      SchedulerBinding.instance.addPostFrameCallback((timeStamp) {
+        selectedItemNotifier.value = widget.initialItem;
+      });
     }
 
-    if (widget.initialItems != oldWidget.initialItems) {
-      selectedItemsNotifier = MultiSelectController(widget.initialItems ?? []);
+    if (widget.initialItems != oldWidget.initialItems &&
+        selectedItemsNotifier.value != widget.initialItems) {
+      SchedulerBinding.instance.addPostFrameCallback((timeStamp) {
+        selectedItemNotifier.value = widget.initialItem;
```

Comment:
This should be `selectedItemsNotifier.value = widget.initialItems ?? [];`

Reply:
In the PR #75, I used `selectedItemsNotifier.value = widget.initialItems ?? [];`. But it raised exception in one of my production case.

**Findings**
- `didUpdateWidget` runs during the build phase. And, updating the state in this phase makes the current widget as dirty and trigger it to be rebuilt. This causes the exception. The same reason why we never call `setState` directly in the build method. 
- Updating the **notifier** value will trigger the **notifyListeners**. Which in turn call the `onChange`. This can happen during the build phase. And if setState is called in `onChange` before build phase completion,  it will throw exception as mentioned above.
- To solve this we can utilize **post frame callbacks** to execute the code after the completion of the build phase. Either we can use `WidgetsBinding` or `SchedulerBinding`. Since, we don't need any lifecycle methods that WidgetsBinding provides, we can only use ScheduleBinding. 

Please note that [WidgetsBinding mixin source](https://github.com/flutter/flutter/blob/main/packages/flutter/lib/src/widgets/binding.dart#L396)

Reply:
Hey @ravindrabarthwal Thanks for the findings, really appreciate it.

Yesterday, I tested the code and this change `selectedItemsNotifier.value = widget.initialItems ?? [];` is a must required one because we are checking for the multi-select items in the `if` condition.

Reply:
@AbdullahChauhan you are correct, that was the miss from my side. I have updated the code, to reflect the same. Appreciate your review.